astropy/io/fits/column.py

```diff
@@ -1364,74 +1364,59 @@ def _guess_format(cls, format, start, dim):
 
         return format, recformat
 
-    def _convert_to_valid_data_type(self, array):
+    def _convert_to_valid_data_type(self, array):  # noqa: PLR0911
```

Comment:
This `noqa` just makes the code less clear for most devs who will have no clue what this means.